optuna/integration/allennlp/_executor.py

```diff
@@ -0,0 +1,237 @@
+import json
+import os
+from typing import Any
+from typing import Dict
+from typing import List
+from typing import Optional
+from typing import Union
+
+import optuna
+from optuna._experimental import experimental
+from optuna._imports import try_import
+from optuna.integration.allennlp._environment import _environment_variables
+
+
+with try_import() as _imports:
+    import allennlp
+    import allennlp.commands
+    import allennlp.common.cached_transformers
+    import allennlp.common.util
+
+# TrainerCallback is conditionally imported because allennlp may be unavailable in
+# the environment that builds the documentation.
+if _imports.is_successful():
+    import _jsonnet
+
+
+_PPID = os.getppid()
+
+"""
+User might want to launch multiple studies that uses `AllenNLPExecutor`.
+Because `AllenNLPExecutor` uses environment variables for communicating
+between a parent process and a child process. A parent process creates a study,
+defines a search space, and a child process trains a AllenNLP model by
+`allennlp.commands.train.train_model`. If multiple processes use `AllenNLPExecutor`,
+the one's configuration could be loaded in the another's configuration.
+To avoid this hazard, we add ID of a parent process to each key of
+environment variables.
+"""
+_PREFIX = "{}_OPTUNA_ALLENNLP".format(_PPID)
+_MONITOR = "{}_MONITOR".format(_PREFIX)
+_PRUNER_CLASS = "{}_PRUNER_CLASS".format(_PREFIX)
+_PRUNER_KEYS = "{}_PRUNER_KEYS".format(_PREFIX)
+_STORAGE_NAME = "{}_STORAGE_NAME".format(_PREFIX)
+_STUDY_NAME = "{}_STUDY_NAME".format(_PREFIX)
+_TRIAL_ID = "{}_TRIAL_ID".format(_PREFIX)
```

Comment:
These variable definitions are duplicated with those in `_pruner.py`. Do you think if it is possible to merge them?

Reply:
Thank you for the suggestion.
ede62aca34f40cc74436e0c83d204ac065e2cbcb creates a module to manage them.

Reply:
Thank you!